// ngx-sharebuttons/esm2015/popup/share-buttons-popup.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
// import { ShareButtonsConfig, SHARE_BUTTONS_CONFIG } from '../../src/public-api';
// import { ShareButtonsModule } from '../../buttons/src/public_api';
// Uncomment the following in development mode
import { SHARE_BUTTONS_CONFIG } from 'ngx-sharebuttons';
import { ShareButtonsModule } from 'ngx-sharebuttons/buttons';
import { ShareButtonsPopup } from './share-buttons-popup';
import { SharePopupService } from './share-popup.service';
import { SharePopupButtonsDirective } from './share-popup-buttons.directive';
import { SharePopupComponent } from './share-popup.component';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
export class ShareButtonsPopupModule {
    static withConfig(config) {
        return {
            ngModule: ShareButtonsModule,
            providers: [{ provide: SHARE_BUTTONS_CONFIG, useValue: config }]
        };
    }
}
ShareButtonsPopupModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    ShareButtonsPopup,
                    SharePopupButtonsDirective,
                    SharePopupComponent
                ],
                imports: [
                    CommonModule,
                    ShareButtonsModule,
                    FontAwesomeModule
                ],
                exports: [
                    ShareButtonsModule,
                    ShareButtonsPopup,
                    SharePopupButtonsDirective,
                    SharePopupComponent
                ],
                providers: [
                    SharePopupService
                ]
            },] }
];
