// @sentry/angular/dist/index.js
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("@sentry/browser"), exports);
var sdk_1 = require("./sdk");
exports.init = sdk_1.init;
var errorhandler_1 = require("./errorhandler");
exports.createErrorHandler = errorhandler_1.createErrorHandler;
exports.SentryErrorHandler = errorhandler_1.SentryErrorHandler;
var tracing_1 = require("./tracing");
exports.getActiveTransaction = tracing_1.getActiveTransaction;
// TODO `instrumentAngularRouting` is just an alias for `routingInstrumentation`; deprecate the latter at some point
exports.instrumentAngularRouting = tracing_1.instrumentAngularRouting;
exports.routingInstrumentation = tracing_1.routingInstrumentation;
exports.TraceClassDecorator = tracing_1.TraceClassDecorator;
exports.TraceMethodDecorator = tracing_1.TraceMethodDecorator;
exports.TraceDirective = tracing_1.TraceDirective;
exports.TraceModule = tracing_1.TraceModule;
exports.TraceService = tracing_1.TraceService;
