// ng2-adsense/bundles/ng2-adsense.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
  typeof define === 'function' && define.amd ? define('ng2-adsense', ['exports', '@angular/core', '@angular/common'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ng2-adsense'] = {}, global.ng.core, global.ng.common));
}(this, (function (exports, core, common) { 'use strict';

  var ADSENSE_TOKEN = new core.InjectionToken('AdsenseConfig');

  var AdsenseComponent = /** @class */ (function () {
      function AdsenseComponent(config, 
      // tslint:disable-next-line:no-any
      platform) {
          this.config = config;
          this.platform = platform;
          /** can be manually set if you need all ads on a page to have same id page-xxx */
          this.adRegion = 'page-' + Math.floor(Math.random() * 10000) + 1;
          /**
           * classes applied to the ins element
           */
          this.className = '';
      }
      AdsenseComponent.prototype.ngOnInit = function () {
          var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o;
          var config = this.config;
          this.adClient = (_a = this.adClient) !== null && _a !== void 0 ? _a : config.adClient;
          this.adSlot = (_b = this.adSlot) !== null && _b !== void 0 ? _b : config.adSlot;
          this.adFormat = (_d = (_c = this.adFormat) !== null && _c !== void 0 ? _c : config.adFormat) !== null && _d !== void 0 ? _d : 'auto';
          this.display = (_f = (_e = this.display) !== null && _e !== void 0 ? _e : config.display) !== null && _f !== void 0 ? _f : 'block';
          this.width = (_g = this.width) !== null && _g !== void 0 ? _g : config.width;
          this.height = (_h = this.height) !== null && _h !== void 0 ? _h : config.height;
          this.layout = (_j = this.layout) !== null && _j !== void 0 ? _j : config.layout;
          this.layoutKey = (_k = this.layoutKey) !== null && _k !== void 0 ? _k : config.layoutKey;
          this.pageLevelAds = (_l = this.pageLevelAds) !== null && _l !== void 0 ? _l : config.pageLevelAds;
          this.adtest = (_m = this.adtest) !== null && _m !== void 0 ? _m : config.adtest;
          this.fullWidthResponsive = (_o = this.fullWidthResponsive) !== null && _o !== void 0 ? _o : config.fullWidthResponsive;
      };
      AdsenseComponent.prototype.ngOnDestroy = function () {
          var iframe = this.ins.nativeElement.querySelector('iframe');
          if (iframe && iframe.contentWindow) {
              iframe.src = 'about:blank';
              iframe.remove();
          }
      };
      AdsenseComponent.prototype.ngAfterViewInit = function () {
          if (common.isPlatformBrowser(this.platform)) {
              this.push();
          }
      };
      AdsenseComponent.prototype.push = function () {
          var p = {};
          if (this.pageLevelAds) {
              p.google_ad_client = this.adClient;
              p.enable_page_level_ads = true;
          }
          if (window) {
              try {
                  // tslint:disable-next-line:no-any
                  (window.adsbygoogle = window.adsbygoogle || []).push(p);
              }
              catch (_a) {
                  // pass
              }
          }
      };
      return AdsenseComponent;
  }());
  AdsenseComponent.decorators = [
      { type: core.Component, args: [{
                  selector: 'ng2-adsense,ng-adsense',
                  template: "\n    <ins\n      #ins\n      class=\"adsbygoogle {{ className }}\"\n      [style.display]=\"display\"\n      [style.width.px]=\"width\"\n      [style.height.px]=\"height\"\n      [attr.data-ad-client]=\"adClient\"\n      [attr.data-ad-slot]=\"adSlot\"\n      [attr.data-ad-format]=\"adFormat\"\n      [attr.data-ad-region]=\"adRegion\"\n      [attr.data-layout]=\"layout\"\n      [attr.data-adtest]=\"adtest\"\n      [attr.data-layout-key]=\"layoutKey\"\n      [attr.data-full-width-responsive]=\"fullWidthResponsive\"\n    >\n    </ins>\n  ",
                  changeDetection: core.ChangeDetectionStrategy.OnPush
              },] }
  ];
  AdsenseComponent.ctorParameters = function () { return [
      { type: undefined, decorators: [{ type: core.Inject, args: [ADSENSE_TOKEN,] }] },
      { type: undefined, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] }
  ]; };
  AdsenseComponent.propDecorators = {
      adClient: [{ type: core.Input }],
      adSlot: [{ type: core.Input }],
      adFormat: [{ type: core.Input }],
      adRegion: [{ type: core.Input }],
      display: [{ type: core.Input }],
      width: [{ type: core.Input }],
      height: [{ type: core.Input }],
      layout: [{ type: core.Input }],
      layoutKey: [{ type: core.Input }],
      pageLevelAds: [{ type: core.Input }],
      adtest: [{ type: core.Input }],
      fullWidthResponsive: [{ type: core.Input }],
      className: [{ type: core.Input }],
      ins: [{ type: core.ViewChild, args: ['ins', { read: core.ElementRef, static: true },] }]
  };

  var AdsenseModule = /** @class */ (function () {
      function AdsenseModule() {
      }
      AdsenseModule.forRoot = function (config) {
          if (config === void 0) { config = {}; }
          return {
              ngModule: AdsenseModule,
              providers: [{ provide: ADSENSE_TOKEN, useValue: config }],
          };
      };
      return AdsenseModule;
  }());
  AdsenseModule.decorators = [
      { type: core.NgModule, args: [{
                  imports: [common.CommonModule],
                  exports: [AdsenseComponent],
                  declarations: [AdsenseComponent],
              },] }
  ];

  /**
   * Generated bundle index. Do not edit.
   */

  exports.ADSENSE_TOKEN = ADSENSE_TOKEN;
  exports.AdsenseComponent = AdsenseComponent;
  exports.AdsenseModule = AdsenseModule;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
